lmdeploy/pytorch/kernels/cuda/fill_kv_cache.py

```diff
@@ -12,6 +14,28 @@ def _div_up(val, other):
     return (val + other - 1) // other
 
 
+@triton.jit
+def _quant_int8(val):
+    val_min = tl.min(val, 1)
+    val_max = tl.max(val, 1)
+    scales = (val_max - val_min) / 255
+    zeros = -val_min / scales
+    q_val = (val / scales[:, None] + zeros[:, None] + 0.5).to(tl.uint8)
+    return q_val, scales, zeros
+
+
+@triton.jit
+def _quant_int4(val1, val2):
+    val_min = tl.minimum(tl.min(val1, 1), tl.min(val2, 1))
```

Comment:
I guess `min(minimum(val1, val2), 1)` might be faster since we only need to perform reduce once.